core/api/container/src/main/java/org/codehaus/cargo/container/spi/deployer/AbstractJsr88Deployer.java

```diff
@@ -423,35 +426,48 @@ else if (DeployableType.WAR.equals(deployable.getType()))
                 + " has no JSR-88 match and cannot be remotely deployed.");
         }
 
-        TargetModuleID targetModule = null;
         TargetModuleID[] modules = deploymentManager.getAvailableModules(moduleType,
             deploymentManager.getTargets());
 
         StringBuilder sb = new StringBuilder();
         sb.append("Cannot find the module \"");
         sb.append(moduleName);
         sb.append("\". Available modules:");
-        for (TargetModuleID module : modules)
-        {
+        
+        List<TargetModuleID> targetModules = new ArrayList<TargetModuleID>();    
```

Comment:
Diamond operator is sufficient, no need for the type

`List<TargetModuleID> targetModules = new ArrayList<>();  `